src/main/java/athena/ui/Ui.java

```diff
@@ -0,0 +1,55 @@
+// Stretch goal: abstract out the messages
+package athena.ui;
+
+import athena.tasks.TaskList;
+
+import java.util.List;
+import java.util.Scanner;
+
+public class Ui {
+    private final TaskList taskList;
+    private final Scanner scanner;
+
+    public Ui(TaskList taskList) {
+        this.taskList = taskList;
+        scanner = new Scanner(System.in);
+    }
+
+    public String readNextLine() {
+        return scanner.nextLine();
+    }
+
+    public void showSpecificTasks(List<Integer> taskNumbers) {
+        for (int taskNumber : taskNumbers) {
+            System.out.printf("%d. %s", taskNumber, taskList.getTaskString(taskNumber));
+            System.out.println();
+        }
```

Comment:
What I did for this is to edit the toString() method of TaskList to print all the tasks, because I personally feel like Ui is a user interface that only handles String and the display of String, so I tried to avoid asking Ui to iterate through TaskList ;P
Mind telling me your rationale too?

Reply:
hmm, because this one was for the `find` command where I only wanted to print out specific tasks, so I thought it made more sense to keep it in the `Ui` class.

Reply:
True true. I solved this problem by creating a new `TaskList` instance containing only filtered tasks and call `toString()` method on it haha